src/toolkit/res_manip.cc

```diff
@@ -65,6 +68,31 @@ std::vector<Resource::Ptr> ResCast(std::vector<Resource::Ptr> rs) {
   return rs;
 }
 
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg) {
+  if (r->quantity() < pkg->fill_min()) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  std::vector<Resource::Ptr> rs;
+  Resource::Ptr r_pkgd;
+  double fill_mass;
+  if (pkg->strategy() == "first") {
+    fill_mass = pkg->fill_max();
+  } else if (pkg->strategy() == "equal") {
+    int num_min_fill = std::floor(r->quantity() / pkg->fill_min());
+    int num_max_fill = std::ceil(r->quantity() / pkg->fill_max());
+    if (num_min_fill >= num_max_fill) {
+      // all material can fit in a package
+      double fill_mass = r->quantity() / num_max_fill;
+    } else {
+      // some material will remain unpackaged, fill up as many max packages as possible
+      fill_mass = pkg->fill_max();
+    }
+  }
+  return fill_mass;
```

Comment:
I suppose, this comment will help clarify the scenario when the condition num_min_fill >= num_max_fill is not true.

```suggestion
  // Handle the scenario when there's leftover material that cannot be packaged
  // due to not enough material for minimum fill, but more than one max filled package worth of material.
  // In this case, fill one package up all the way and return the remaining to the buffer.
  return fill_mass;
```

Reply:
Hi @ahnaf-tahmid-chowdhury, thanks for your comment! This method doesn't actually handle the repackaging, it only calculates how much material will go in the package(s), so this description isn't quite right for what this part of the code is doing. Also, a comment describing the functionality should be in the header file at the top, rather than at the bottom of the source. 

I'm going to commit your suggestion because I appreciate you trying help clarify, and I'll make the changes I just described in my next PR so there's better description in the header for `GetFillMass` and `Repackage`